platforms/core-runtime/build-configuration/src/test/groovy/org/gradle/internal/buildconfiguration/DaemonJvmPropertiesAccessorTest.groovy

```diff
@@ -34,6 +34,7 @@ class DaemonJvmPropertiesAccessorTest extends Specification {
         def properties = [
             "toolchainVersion": "12",
             "toolchainVendor": "BELLSOFT",
+            "toolchainNativeImage": "true",
```

Comment:
To me this suggest that property is expecting a native image path or similar instead of a boolean. Potentially considering replacing with  `toolchainRequiresNativeImage`, `toolchainNativeImageRequired` or similar to improve readability